lms/djangoapps/learner_home/views.py

```diff
@@ -238,10 +238,24 @@ def get_ecommerce_payment_page(user):
 @function_trace("get_cert_statuses")
 def get_cert_statuses(user, course_enrollments):
     """Get cert status by course for user enrollments"""
-    return {
-        enrollment.course_id: cert_info(user, enrollment)
-        for enrollment in course_enrollments
-    }
+
+    cert_statuses = {}
+
+    for enrollment in course_enrollments:
+        # APER-2171 - trying to get a cert for a deleted course can throw an exception
+        # Wrap in exception handling to avoid this issue.
+        try:
+            certificate_for_course = cert_info(user, enrollment)
+
+            if certificate_for_course:
+                cert_statuses[enrollment.course_id] = certificate_for_course
+
+        except Exception as ex:  # pylint: disable=broad-except
```

Comment:
I'm a little hesitant so v. broadly squash exceptions. Is there a base class for the exceptions we want to capture?  Can we build a small tuple of them?

Reply:
The original exception is a `TypeError` but I'd prefer to catch everything, overly narrow exception handling is what _caused_ this issue originally.

The top-level view doesn't have a highest-level error case so any fall-through exception will kill the entire page. If you can think of a reason we would _want_ to kill the entire page load if there's a more exotic exception, I'm happy to make this more targeted but until then, I think this is a safer approach.